js/index.js

```diff
@@ -0,0 +1,163 @@
+productList.className = "product-list row pt-12 sm-p-reset";
+
+if (productsData.length > 0) {
+  //add event listeners to show and hide button
+
+  productsData.forEach((product) => {
+    //create product item
+    var productItem = document.createElement("li");
+    productItem.className = "product-item col col-3 col-sm-6";
+
+    //create product link
+    var productLink = document.createElement("a");
+    productLink.className = "product-link";
+
+    var productImageWrapper = document.createElement("div");
+    productImageWrapper.className = "relative product-image-wrapper";
+
+    //create product image
+    var productImage = document.createElement("img");
+    productImage.className = "product-img";
+    productImage.src = product.image;
+    productImage.alt = product.name;
+
+    //add event listeners to show and hide button
+
+    //create product name
+    var productName = document.createElement("h4");
+    productName.className = "product-name";
+    productName.textContent = product.name;
+
+    //create product price group
+    var productPriceGroup = document.createElement("div");
+    productPriceGroup.className =
+      "product-price-group d-flex justify-space-between";
+
+    //create product price
+    var productPrice = document.createElement("span");
+    productPrice.textContent = product.price;
+    productPrice.className = "product-price";
+
+    //check if product has discount
+    if (product.discount > 0) {
+      productDiscount(
+        product.discount,
+        productLink,
+        productPriceGroup,
+        productPrice,
+        product
+      );
+    }
+
+    productList.appendChild(productItem);
+    productItem.appendChild(productLink);
+    productImageWrapper.appendChild(productImage);
+    productLink.appendChild(productImageWrapper);
+    productLink.appendChild(productName);
+    productLink.appendChild(productPriceGroup);
+
+    productPriceGroup.appendChild(productPrice);
+
+    var btnAddToCart = document.createElement("button");
+    btnAddToCart.className = "btn btn-primary btn-add-to-cart absolute";
+    btnAddToCart.textContent = "Add to cart";
+    btnAddToCart.addEventListener("click", function (e) {
+      var cartItems = JSON.parse(localStorage.getItem("cartItems"));
+
+      var productToCart = {
+        id: product?.id,
+        name: product?.name,
+        price: product?.price,
+        quantity: 1,
+      };
+
+      if (cartItems !== null) {
+        var productExists = cartItems.find(
+          (item) => item.id === productToCart.id
+        );
+
+        if (!productExists) {
+          cartItems.push(productToCart);
+          localStorage.setItem("cartItems", JSON.stringify(cartItems));
+          return;
+        } else {
+          var cartItemToUpdate = cartItems.find(function (item) {
+            return item.id === productToCart.id;
+          });
+
+          cartItemToUpdate.quantity += 1;
+
+          localStorage.setItem("cartItems", JSON.stringify(cartItems));
```

Comment:
Dòng code này dùng chung cho 2 case productExists or not => Tách nó ra dùng chung, ko để vào trong if, else e nha